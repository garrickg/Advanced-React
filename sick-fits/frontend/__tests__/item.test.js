import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ItemComponent from '../components/item';

const testItem = {
  id: 'ABC123',
  title: 'A Test Item',
  price: 5000,
  description: 'Just a fake item for testing',
  image: 'test.jpg',
  largeImage: 'largeTest.jpg',
};

describe('<Item />', () => {
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(<ItemComponent item={testItem} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  // it('renders and displays properly', () => {
  //   const wrapper = shallow(<ItemComponent item={testItem} />);
  //   const PriceTag = wrapper.find('PriceTag');
  //   expect(PriceTag.children().text()).toBe('$50');
  //   expect(wrapper.find('Title a').text()).toBe(testItem.title);
  //   const img = wrapper.find('img');
  //   expect(img.props().src).toBe(testItem.image);
  //   expect(img.props().alt).toBe(testItem.title);
  // });

  // it('renders out the button properly', () => {
  //   const wrapper = shallow(<ItemComponent item={testItem} />);
  //   const ButtonList = wrapper.find('.buttonList');
  //   expect(ButtonList.children()).toHaveLength(3);
  //   expect(ButtonList.find('Link')).toHaveLength(1);
  //   expect(ButtonList.find('AddToCart')).toHaveLength(1);
  //   expect(ButtonList.find('DeleteItem')).toHaveLength(1);
  // });
});
