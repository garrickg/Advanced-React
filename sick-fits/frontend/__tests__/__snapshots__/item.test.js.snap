// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matches the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A Test Item"
    src="test.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        A Test Item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $50
  </PriceTag>
  <p>
    Just a fake item for testing
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit...
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeleteItem
      id="ABC123"
    >
      Delete Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
